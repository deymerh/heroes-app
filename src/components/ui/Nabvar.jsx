import React, { useContext } from 'react';
import { Link, NavLink, useHistory } from 'react-router-dom'
import { AuthContext } from '../../auth/AuthContext';
import { types } from '../../types/types';

export const Navbar = () => {
  let history = useHistory();
  const { user: { name }, dispatch } = useContext(AuthContext);
  const logunt = () => {
    dispatch({
      type: types.logout
    });
    history.replace('/login');
  };
  return (
    <nav className="navbar navbar-expand-lg navbar-dark bg-dark">
      <Link className="navbar-brand" to="/">Asociaciones</Link>

      <div className="navbar-collapse">
        <div className="navbar-nav">
          <NavLink activeClassName="active" className="nav-item nav-link" exact to="/marvel" >Marvel</NavLink>
          <NavLink activeClassName="active" className="nav-item nav-link" exact to="/dc">DC</NavLink>
          <NavLink activeClassName="active" className="nav-item nav-link" exact to="/search">Search</NavLink>
        </div>
      </div>
      <ul className="navbar-nav ml-auto">
        <span className="nav-item nav-link text-info">{name}</span>
        <button className="nav-item nav-link btn" onClick={logunt}> Logout</button>
      </ul>
    </nav>
  )
}