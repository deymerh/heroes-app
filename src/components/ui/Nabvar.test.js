import '@testing-library/jest-dom';
import { mount } from 'enzyme';
import { MemoryRouter, Router } from 'react-router-dom';
import { AuthContext } from '../../auth/AuthContext';
import { types } from '../../types/types';
import { Navbar } from './Nabvar';

describe('Pruebas en el componente <Nabvar/>', () => {
  const historyMock = {
    pathname: '',
    location: {},
    replace: jest.fn(),
    listen: jest.fn(),
    createHref: jest.fn() 
  }
  const ContextValue = {
    dispatch: jest.fn(),
    user:{
      logged: true,
      name: 'Deymer'
    }
  }
  const wrapper = mount(
    <AuthContext.Provider value={ContextValue}>
      <MemoryRouter>
        <Router history={historyMock}>
          <Navbar />
        </Router>
      </MemoryRouter>
    </AuthContext.Provider>
  );
  afterEach(()=>{
    jest.clearAllMocks();
  })
  test('Debe mostrarse correctamente', () => {
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('.text-info').text().trim()).toBe('Deymer');  
  });

  test('Debe llamar al logout y usar el history.replace', () => {
    wrapper.find('button').prop('onClick')();
    expect(ContextValue.dispatch).toHaveBeenCalledWith({type: types.logout});
    expect(historyMock.replace).toHaveBeenCalledWith('/login');
  });
  
});
