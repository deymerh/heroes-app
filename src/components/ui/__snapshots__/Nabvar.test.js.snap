// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <Nabvar/> Debe mostrarse correctamente 1`] = `
<nav
  className="navbar navbar-expand-lg navbar-dark bg-dark"
>
  <a
    className="navbar-brand"
    onClick={[Function]}
  >
    Asociaciones
  </a>
  <div
    className="navbar-collapse"
  >
    <div
      className="navbar-nav"
    >
      <a
        aria-current={null}
        className="nav-item nav-link"
        onClick={[Function]}
      >
        Marvel
      </a>
      <a
        aria-current={null}
        className="nav-item nav-link"
        onClick={[Function]}
      >
        DC
      </a>
      <a
        aria-current={null}
        className="nav-item nav-link"
        onClick={[Function]}
      >
        Search
      </a>
    </div>
  </div>
  <ul
    className="navbar-nav ml-auto"
  >
    <span
      className="nav-item nav-link text-info"
    >
      Deymer
    </span>
    <button
      className="nav-item nav-link btn"
      onClick={[Function]}
    >
       Logout
    </button>
  </ul>
</nav>
`;
